pallets/creator-staking/src/types/impls.rs

```diff
@@ -169,10 +169,8 @@ impl<Balance, MaxEraStakeValues> StakesInfo<Balance, MaxEraStakeValues>
     pub(crate) fn claim(&mut self) -> (EraIndex, Balance) {
         if let Some(oldest_era_stake) = self.stakes.first() {
             let oldest_era_stake = *oldest_era_stake;
-            let has_no_stake_updates_for_the_next_era =
-                oldest_era_stake.era + 2 <= self.stakes[1].era;
 
-            if self.stakes.len() == 1 || has_no_stake_updates_for_the_next_era {
+            if self.stakes.len() == 1 || oldest_era_stake.era + 2 <= self.stakes[1].era {
```

Comment:
tbh need to add comment why `+ 2`, I read the code and I have forgot why :)

Reply:
There was a comment right below. Moved it above the `if` statement and changed the text to make it more clear.